client/service/src/metrics.rs

```diff
@@ -232,5 +289,50 @@ impl MetricsService {
 				);
 			}
 		}
+
+		// Update/send network status information, if any.
+		if let Some(net_status) = net_status {
+			let num_peers = net_status.num_connected_peers;
+			let total_bytes_inbound = net_status.total_bytes_inbound;
+			let total_bytes_outbound = net_status.total_bytes_outbound;
+
+			let diff_bytes_inbound = total_bytes_inbound - self.last_total_bytes_inbound;
+			let diff_bytes_outbound = total_bytes_outbound - self.last_total_bytes_outbound;
+			let (avg_bytes_per_sec_inbound, avg_bytes_per_sec_outbound) =
+				if elapsed > 0 {
+					self.last_total_bytes_inbound = total_bytes_inbound;
+					self.last_total_bytes_outbound = total_bytes_outbound;
+					(diff_bytes_inbound / elapsed, diff_bytes_outbound / elapsed)
+				} else {
+					(diff_bytes_inbound, diff_bytes_outbound)
+				};
+
+			telemetry!(
+				SUBSTRATE_INFO;
+				"system.interval";
+				"peers" => num_peers,
+				"bandwidth_download" => avg_bytes_per_sec_inbound,
+				"bandwidth_upload" => avg_bytes_per_sec_outbound,
+			);
```

Comment:
I didn't realize when reviewing, but I think that the telemetry server expects the message to be full. I don't think it's possible to send partial information.
In practice, at the moment, the number of peers and bandwidth stay at 0 on telemetry for nodes that have this PR in.

Reply:
Or maybe the problem is caused by `net_status` never being `Some`, I'm not sure.

Reply:
>  I don't think it's possible to send partial information.

In principle it does, mostly. I opened https://github.com/paritytech/substrate-telemetry/pull/289.